modules/common/services/audio.nix

```diff
@@ -91,17 +91,70 @@ in
     # Start pipewire on system boot
     systemd.services.pipewire.wantedBy = [ "multi-user.target" ];
 
-    systemd.services."pulseaudio-set-profile" =
+    systemd.services."initialize-audio-profile" =
       let
-        pulse-set-profile = pkgs.writeShellScriptBin "pulseaudio-set-profile" ''
-          ${pkgs.pulseaudio}/bin/pactl set-card-profile "alsa_card.pci-0000_00_07.0-platform-skl_hda_dsp_generic" "HiFi (Headphones, Mic1, Mic2)"
-          ${pkgs.pulseaudio}/bin/pactl set-card-profile "alsa_card.pci-0000_00_07.0-platform-skl_hda_dsp_generic" "HiFi (Mic1, Mic2, Speaker)"
-        '';
+        initialize-audio-profile = pkgs.writeShellApplication {
+          name = "initialize-audio-profile";
+          runtimeInputs = [
+            pkgs.pulseaudio
+            pkgs.jq
+          ];
+          text = ''
+            function setProfile() {
+                if [[ $# == 2 ]] && [[ -n "$1" ]] && [[ -n "$2" ]]; then
```

Comment:
It should be enough to check that there's exactly two arguments `if [[ $# -eq 2 ]]; then`

Reply:
I agree that the checks overlap but i would rather keep the -n checks to catch empty arguments